src/Apps/Order2/Routes/Checkout/Components/PaymentStep/Order2PaymentForm.tsx

```diff
@@ -117,106 +122,133 @@ const PaymentFormContent: React.FC<PaymentFormContentProps> = ({ order }) => {
     },
   }
 
-  const onChange = event => {
-    const { elementType, collapsed } = event
+  const onChange = StripePaymentElementChangeEvent => {
+    const { elementType, collapsed } = StripePaymentElementChangeEvent
     if (elementType === "payment" && !collapsed) {
-      setSelectedPaymentMethod("new")
+      setSelectedPaymentMethod("stripe")
     }
   }
 
   const onClickSavedPaymentMethods = () => {
+    setErrorMessage(null) // Clear any previous error messages
     setSelectedPaymentMethod("saved")
     elements?.getElement("payment")?.collapse()
   }
 
   const onClickWirePaymentMethods = () => {
+    setErrorMessage(null) // Clear any previous error messages
     setSelectedPaymentMethod("wire")
     elements?.getElement("payment")?.collapse()
   }
 
-  // From Stripe docs
   const handleError = error => {
     setErrorMessage(error.message)
+    setIsSubmittingToStripe(false)
   }
 
-  // From Stripe docs
   const handleSubmit = async event => {
     event.preventDefault()
-    setIsSubmittingToStripe(true)
 
-    const { error: submitError } = await elements.submit()
-    if (submitError) {
-      logger.error(submitError)
-      handleError(submitError)
+    if (selectedPaymentMethod === "none") {
+      setSubtitleErrorMessage("Select a payment method")
       return
     }
```

Comment:
Similar to the other comment about undefined vs null, I would make this `null | "saved" | "stripe" | "wire"` in the useState up above. Then you can treat null as the value and use the more idiomatic `if (!selectedPaymentMethod) {...`